R/opts.R

```diff
@@ -0,0 +1,447 @@
+  stop_if_not_gt(data = data)
+
+  align <- match.arg(align)
+
+  tab_options(data, heading.align = align)
+}
+
+#' Option to use all caps in select table locations
+#'
+#' Sometimes an all-capitalized look is suitable for a table. With the
+#' `opt_all_caps()` function, we can transform characters in the column labels,
+#' the stub, and in all row groups in this way (and there's control over which
+#' of these locations are transformed).
+#'
+#' This function serves as a convenient shortcut for `<gt_tbl> %>%
+#' tab_options(<location>.text_transform = "uppercase", <location>.font.size =
+#' pct(80), <location>.font.weight = "bolder")` (for all `locations` selected).
+#'
+#' @inheritParams fmt_number
+#' @param all_caps A logical value to indicate whether the text transformation
+#'   to all caps should be performed (`TRUE`, the default) or reset to default
+#'   values (`FALSE`) for the `locations` targeted.
+#' @param locations Which locations should undergo this text transformation? By
+#'   default it includes all of the `"column_labels"`, the `"stub"`, and the
+#'   `"row_group"` locations. However, we could just choose one or two of those.
+#'
+#' @return An object of class `gt_tbl`.
+#'
+#' @examples
+#' # Use `exibble` to create a gt table with
+#' # a number of table parts added; all text
+#' # in the column labels, the stub, and in
+#' # all row groups is to be transformed to
+#' # all caps using `opt_all_caps()`
+#' tab_1 <-
+#'   exibble %>%
+#'   gt(rowname_col = "row", groupname_col = "group") %>%
+#'   summary_rows(
+#'     groups = "grp_a",
+#'     columns = vars(num, currency),
+#'     fns = list(
+#'       min = ~min(., na.rm = TRUE),
+#'       max = ~max(., na.rm = TRUE)
+#'     )) %>%
+#'   grand_summary_rows(
+#'     columns = vars(currency),
+#'     fns = list(
+#'       total = ~sum(., na.rm = TRUE)
+#'     )) %>%
+#'   tab_source_note(source_note = "This is a source note.") %>%
+#'   tab_footnote(
+#'     footnote = "This is a footnote.",
+#'     locations = cells_body(columns = 1, rows = 1)
+#'   ) %>%
+#'   tab_header(
+#'     title = "The title of the table",
+#'     subtitle = "The table's subtitle"
+#'   ) %>%
+#'   opt_all_caps()
+#'
+#' @section Figures:
+#' \if{html}{\figure{man_opt_all_caps_1.svg}{options: width=100\%}}
+#'
+#' @family Table Option Functions
+#' @section Function ID:
+#' 9-4
+#'
+#' @export
+opt_all_caps <- function(data,
+                         all_caps = TRUE,
+                         locations = c("column_labels", "stub", "row_group")) {
+
+  # Perform input object validation
+  stop_if_not_gt(data = data)
+
+  # Ensure that all named locations are valid
+  if (!all(locations %in% c("column_labels", "stub", "row_group"))) {
+    stop("The available locations are `column_labels`, `stub`, and `row_group`.",
+         call. = FALSE)
+  }
+
+  # Get vector of `tab_options()` arg names for all specific `locations`
+  options_vec <- table_all_caps_vec(locations)
```

Comment:
Inline the logic from `table_all_caps_vec` and delete that function; the logic is tightly coupled to this, and you're only using it once.

Reply:
Done